views.py

```diff
@@ -1240,19 +1241,17 @@ def typesetting_proofreading_assignment(request, assignment_id):
     if request.POST:
         form = forms.ProofingForm(request.POST, instance=assignment)
 
-        if form.is_valid():
-            form.save()
-
         if request.FILES:
             logic.handle_proofreader_file(
                 request,
                 assignment,
                 assignment.round.article,
             )
 
-        if 'complete' in request.POST:
-            unproofed_galleys = assignment.unproofed_galleys(galleys)
-            if not unproofed_galleys:
+        if form.is_valid():
+            form.save()
+
+            if form.is_confirmed():
```

Comment:
There is a slight change in behaviour here:

The check for unproofed galleys used to stop an author from completing an assignment. With this change, the author is warned but can still override the system. Nothing necessarily wrong  - I know you are a fan of user control, but just wanted to double check that this new behaviour is desired.

Reply:
Ah, I just saw this previous comment: https://github.com/BirkbeckCTP/typesetting/pull/160#issuecomment-1189074969